chapters/sampling-randomization-power.tex

```diff
@@ -1,391 +1,434 @@
-Note that there are two distinct concepts referred to here by ``randomization'':
-the conceptual process of assigning units to treatment arms,
-and the technical process of assigning random numbers in statistical software,
-which is a part of all tasks that include a random component.\sidenote{\url{https://blog.stata.com/2016/03/10/how-to-generate-random-numbers-in-stata/}}
-If the randomization seed for the statistical software is not set,
-then its pseudorandom algorithm will pick up where it left off.
-By setting the seed, you force it to restart from the same point.
-In Stata, \texttt{set seed [seed]} will accomplish this.
-
-The code below code loads and sets up the \texttt{auto.dta} dataset
-for any random process. Note the three components: versioning, sorting, and seeding.
-Why are \texttt{check1} and \texttt{check3} the same? Why is \texttt{check2} different?
-
-\codeexample{replicability.do}{./code/replicability.do}
-
-Commands like \texttt{bys:} and \texttt{merge} will re-sort your data as part of their execution,
-To reiterate: any process that includes a random component
-is a random process, including sampling, randomization, power calculation,
-and many algorithms like bootstrapping.
-and other commands may alter the seed without you realizing it.\sidenote{\url{https://dimewiki.worldbank.org/wiki/Randomization_in_Stata}}
-Any of these things will cause the output to fail to replicate.
-Therefore, each random process should be independently executed
-to ensure that these three rules are followed.
-Before shipping the results of any random process,
+Because numbers are assigned to each observation in sequence,
+changing their order will change the result of the process.
+A corollary is that the underlying data must be unchanged between runs:
+you must make a fixed final copy of the data when you run a randomization for fieldwork.
+In Stata, the only way to guarantee a unique sorting order is to use
+\texttt{isid [id\_variable], sort}. (The \texttt{sort, stable} command is insufficient.)
+You can additionally use the \texttt{datasignature} command to make sure the 
+data is unchanged.
+
+\textbf{Seeding} means manually setting the start-point of the randomization algorithm.
+You can draw a six-digit seed randomly by visiting \url{http://bit.ly/stata-random}.
+(This link is a shortcut to a specific request on \url{https://www.random.org}.)
+There are many more seeds possible but this is a large enough set for most purposes.
+In Stata, \texttt{set seed [seed]} will set the generator to that state.
+You should use exactly one seed per randomization process.
+The most important thing is that each of these seeds is truly random,
+so do not use shortcuts such as the current date or a fixed seed.
+You will see in the code below that we include the source and timestamp for verification.
+Any process that includes a random component is a random process,
+including sampling, randomization, power calculation, and algorithms like bootstrapping.
```

Comment:
not all _power calculation_ has a random component. Do we want to say _power calculation simulations_ here?

Reply:
yep b48a5c0dbe019bbc1d70b3cabe33fe938fecabc9